MekHQ/src/mekhq/campaign/personnel/familyTree/Genealogy.java

```diff
@@ -118,60 +108,64 @@ public void setSpouse(@Nullable UUID spouse) {
      * @param formerSpouse a former spouse to add the the current person's list
      */
     public void addFormerSpouse(FormerSpouse formerSpouse) {
-        getFormerSpouses().add(formerSpouse);
+        getFormerSpouses().add(Objects.requireNonNull(formerSpouse));
     }
 
     /**
-     * This is implemented for future use, as it will be required for the family tree implementation
-     * @param formerSpouseId the former spouse's id to remove from the current person's list
+     * @param formerSpouse the former spouse to remove from the current person's list
      */
-    public void removeFormerSpouse(UUID formerSpouseId) {
-        getFormerSpouses().removeIf(formerSpouse -> formerSpouse.getFormerSpouseId().equals(formerSpouseId));
+    public void removeFormerSpouse(Person formerSpouse) {
+        getFormerSpouses().removeIf(ex -> ex.getFormerSpouse().equals(formerSpouse));
     }
 
     /**
      * @return the family map for this person
      */
-    public Map<FamilialRelationshipType, List<UUID>> getFamily() {
+    public Map<FamilialRelationshipType, List<Person>> getFamily() {
         return family;
     }
 
+    /**
+     * @param family the new family map for this person
+     */
+    public void setFamily(Map<FamilialRelationshipType, List<Person>> family) {
+        this.family = family;
+    }
+
     /**
      * This is used to add a new family member
      * @param relationshipType the relationship type between the two people
-     * @param id the id of the person to add
+     * @param person the person to add
      */
-    public void addFamilyMember(FamilialRelationshipType relationshipType, UUID id) {
-        if (id != null) {
+    public void addFamilyMember(FamilialRelationshipType relationshipType, @Nullable Person person) {
+        if (person != null) {
             getFamily().putIfAbsent(relationshipType, new ArrayList<>());
```

Comment:
Is getFamily() guaranteed to be non-null?

Reply:
It is. Anything that is nullable is marked as such.